src/ScriptEngine/ScriptEngine.cs

```diff
@@ -76,22 +76,39 @@ void LoadDLL(string path, GameObject obj)
                     dll.Write(ms);
                     var ass = Assembly.Load(ms.ToArray());
 
-                    foreach(Type type in ass.GetTypes())
+                    try
                     {
-                        if(typeof(BaseUnityPlugin).IsAssignableFrom(type))
+                        foreach (Type type in ass.GetTypes())
                         {
-                            var metadata = MetadataHelper.GetMetadata(type);
-                            if(metadata != null)
+                            if (typeof(BaseUnityPlugin).IsAssignableFrom(type))
                             {
-                                var typeDefinition = dll.MainModule.Types.First(x => x.FullName == type.FullName);
-                                var typeInfo = Chainloader.ToPluginInfo(typeDefinition);
-                                Chainloader.PluginInfos[metadata.GUID] = typeInfo;
-
-                                Logger.Log(LogLevel.Info, $"Reloading {metadata.GUID}");
-                                StartCoroutine(DelayAction(() => obj.AddComponent(type))); 
+                                var metadata = MetadataHelper.GetMetadata(type);
+                                if (metadata != null)
+                                {
+                                    var typeDefinition = dll.MainModule.Types.First(x => x.FullName == type.FullName);
+                                    var typeInfo = Chainloader.ToPluginInfo(typeDefinition);
+                                    Chainloader.PluginInfos[metadata.GUID] = typeInfo;
+
+                                    Logger.Log(LogLevel.Info, $"Reloading {metadata.GUID}");
+                                    StartCoroutine(DelayAction(() => obj.AddComponent(type)));
+                                }
                             }
                         }
                     }
+                    catch (ReflectionTypeLoadException e)
+                    {
+                        var sbMessage = new StringBuilder();
+                        sbMessage.AppendLine($"Error While loading {path}");
+                        sbMessage.AppendLine("-- Successfully Read Types --");
+                        foreach (var t in e.Types)
+                            if (t != null) sbMessage.AppendLine(t.ToString());
+                        sbMessage.AppendLine("\r\n-- LoaderExceptions --");
+                        foreach (var l in e.LoaderExceptions)
+                            if (l != null) sbMessage.AppendLine(l.ToString());
```

Comment:
Is there a need for a null check here? I know that `e.Types` can contain null entries, but I'm fairly certain `e.LoaderExceptions` has no nulls. 

For example, in [`TypeLoader.TypeLoadExceptionToString`](https://github.com/BepInEx/BepInEx/blob/master/BepInEx/Bootstrap/TypeLoader.cs#L232-L259) that we use for printing `ReflectionTypeLoadException` in BepInEx itself we don't have such a check.

Reply:
I'll follow. I wondered if `e.LoaderExceptions` has nulls, but chose safe way without enough survey ^^;